### PR TITLE
Add sales report endpoint

### PR DESCRIPTION
**Pull request by matti-lamppu**
# Hitas Pull Request

# Description

Adds endpoint for creating and downloading sales report Excel file.

## Pull request checklist

Check the boxes for each DoD item that has been completed:

- **Testing**
  - [x] Changes have been tested
  - [x] Automatic tests have been added
- **Database**
  - [ ] Database migrations will work in the DEV & TEST environments
  - [ ] initial.json has been updated to work with migrations
  - [ ] Oracle migration has been updated
- **Documentation**
  - [ ] Tooltips have been added in the frontend for all new fields
  - [x] OpenAPI definitions have been updated
  - [ ] Test instructions have been written for the customer in the appropriate ticket in Jira
  - [ ] Terminology page in Confluence has been updated

## Test plan

- [x] Automated tests
- [ ] Uncomment the lines below from the top of `backend/openapi.yaml` and open the Swagger interface at `localhost:8090`. Add `http://localhost:8090` to `CORS_ALLOWED_ORIGINS` in `backend/.env`. Add `authentication_classes = []` & `permission_classes = []` to `hitas.views.reports.SalesReportView`. Use the new report endpoint from the Swagger interface while the backend is running at `localhost:8000` and observe the downloaded Excel file.

```yaml
#servers:
#  - url: http://localhost:8000
#    description: Local development server
```

## Tickets

This pull request resolves all or part of the following ticket(s): HT-606

### CODE REVIEW

File: backend/hitas/views/reports.py
Comment (ranta) [inline backend/hitas/views/reports.py:28]: Don't all of our other PDF endpoints use GET instead of POST? 🤔 I'd like to keep things consistent if possible :)

Comment (matti-lamppu) [reply backend/hitas/views/reports.py:28]: This is an Excel endpoint, but `download-regulation-results` Excel endpoint also uses GET, so I can convert this to get for consistency. 

Although this endpoint has all of its parameters required when the other endpoint has only optional parameters which suit query parameters better..

Comment (matti-lamppu) [reply backend/hitas/views/reports.py]: Done

**Comment by matti-lamppu** [comment]
Still needs approval from the customer before merging, unless we wanna do an update later

File: backend/hitas/views/reports.py
Comment (ranta) [inline backend/hitas/views/reports.py:21]: 🤔

Comment (ranta) [reply backend/hitas/views/reports.py:21]: We can get this merged for now, and fix it in some later PR to avoid rebasing all of them 😄

File: backend/hitas/services/reports.py
Comment (ranta) [inline backend/hitas/services/reports.py:74]: I would change the "Ei Tilastoihin" value from "TRUE/FALSE" to something more descriptive. e.g. Only showing TRUE values

**Review by ranta — APPROVED** [review]
I have some formatting issues with dates, but that most likely just a mac Sheets app problem 😅

File: backend/hitas/views/reports.py
Comment (matti-lamppu) [reply backend/hitas/views/reports.py:21]: Isn't this correct? We are giving a range to search sales from, so the start of the range needs to be before the end of the range (or the same, if you just want one day).

File: backend/hitas/services/reports.py
Comment (matti-lamppu) [reply backend/hitas/services/reports.py:74]: I can put X / " " there also.

Comment (matti-lamppu) [reply backend/hitas/services/reports.py:74]: Actually, we changed this already so that excluded sales are not in this report, based on the customer's feedback, so this column should be removed.

Comment (matti-lamppu) [reply backend/hitas/services/reports.py:74]: Field removed

File: backend/hitas/views/reports.py
Comment (ranta) [reply backend/hitas/views/reports.py:21]: Yes nvm, brain fart 😅

**Pull request merged**
